source/jormungandr/jormungandr/scenarios/distributed.py

```diff
@@ -90,8 +90,17 @@ def _compute_all(self, future_manager, request, instance, krakens_call, context)
         logger = logging.getLogger(__name__)
         logger.debug('request datetime: %s', request['datetime'])
 
-        requested_dep_modes = {mode for mode, _, _ in krakens_call}
-        requested_arr_modes = {mode for _, mode, _ in krakens_call}
+        requested_dep_modes_with_pt = {
+            mode for mode, _, direct_path_type in krakens_call if direct_path_type != "only"
+        }
+        requested_arr_modes_with_pt = {
+            mode for _, mode, direct_path_type in krakens_call if direct_path_type != "only"
+        }
+
+        requested_direct_path_modes = {
+            mode for mode, _, direct_path_type in krakens_call if direct_path_type == "only"
+        }
+        requested_direct_path_modes.update(requested_dep_modes_with_pt)
```

Comment:
Not sure at all : Aren't you supposed to do that only when requested_direct_path_modes is empty?

Reply:
We need to compute direct_paths for modes in first_sections_modes AND modes in direct_path_modes